docs/current_docs/cookbook/snippets/execute-concurrently/go/main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"dagger/my-module/internal/dagger"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type MyModule struct{}
+
+// Return the result of running unit tests
+func (m *MyModule) Test(ctx context.Context, source *dagger.Directory) (string, error) {
+	return m.BuildEnv(source).
+		WithExec([]string{"npm", "run", "test:unit", "run"}).
+		Stdout(ctx)
+}
+
+// Return the result of running the linter
+func (m *MyModule) Lint(ctx context.Context, source *dagger.Directory) (string, error) {
+	return m.BuildEnv(source).
+		WithExec([]string{"npm", "run", "lint"}).
+		Stdout(ctx)
+}
+
+// Return the result of running the type-checker
+func (m *MyModule) Typecheck(ctx context.Context, source *dagger.Directory) (string, error) {
+	return m.BuildEnv(source).
+		WithExec([]string{"npm", "run", "type-check"}).
+		Stdout(ctx)
+}
+
+// Run linter, type-checker, unit tests concurrently
+func (m *MyModule) RunAllTests(ctx context.Context, source *dagger.Directory) (string, error) {
+	var testResult, lintResult, typecheckResult string
+	var testErr, lintErr, typecheckErr error
+
+	// Create error group
+	eg, gctx := errgroup.WithContext(ctx)
+
+	// Run linter
+	eg.Go(func() error {
+		lintResult, lintErr = m.Lint(gctx, source)
+		return lintErr
+	})
+
+	// Run type-checker
+	eg.Go(func() error {
+		typecheckResult, typecheckErr = m.Typecheck(gctx, source)
+		return typecheckErr
+	})
+
+	// Run unit tests
+	eg.Go(func() error {
+		testResult, testErr = m.Test(gctx, source)
+		return testErr
+	})
+
+	// Wait for all tests to complete
+	// If any test fails, return the error
+	if err := eg.Wait(); err != nil {
+		return "", fmt.Errorf("error: %w", err)
+	}
+
+	// If all tests succeed, print the test results
+	return testResult + lintResult + typecheckResult, nil
```

Comment:
Might be worth doing `strings.Join([]string{testResult, lintResult, typecheckResult}, "\n")` to intersperse with newlines?